### PR TITLE
chore: Update Prometheus dependency

### PR DESCRIPTION
**Pull request by SuperQ**
Update Prometheus from v2.33.5 to v2.36.2.

Signed-off-by: SuperQ <superq@gmail.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

**Review by kakkoyun — COMMENTED** [review]
This looks good to me. 

I think we need to upgrade Cortex. The cyclic dependency started to get annoying. I've been facing similar issues. I think we can consider vendoring Cortex query frontend to break the cycles in the short-term and we can create it in the long-term.

And could you please add a changelog entry?

**Comment by SuperQ** [comment]
Yes, I started working on the cortex upgrade as well. But Cortex first needs to upgrade their use of Thanos.

**Comment by fpetkovski** [comment]
Looks like Thanos was updated in Cortex: https://github.com/cortexproject/cortex/pull/4780

**Comment by SuperQ** [comment]
Ok, this should be much closer now.

**Review by kakkoyun — DISMISSED** [review]
LGTM

**Review by kakkoyun — DISMISSED** [review]

**Review by kakkoyun — DISMISSED** [review]

**Comment by kakkoyun** [comment]
To fix e2e tests https://github.com/thanos-io/thanos/pull/5506

**Review by kakkoyun — APPROVED** [review]

**Review by matej-g — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 - [#5447](https://github.com/thanos-io/thanos/pull/5447) Promclient: Ignore 405 status codes for Prometheus buildVersion requests
 - [#5451](https://github.com/thanos-io/thanos/pull/5451) Azure: Reduce memory usage by not buffering file downloads entirely in memory.
+- [#5484](https://github.com/thanos-io/thanos/pull/5484) Update Prometheus deps to v2.36.2.
```
[the remark this answers is not in the document]
Comment (kakkoyun) [reply CHANGELOG.md:31]: True, it's not user-facing, but as far as I can recall we mention Prometheus updates in the changelog.

**Pull request merged**